app/src/pages/Play/useSecondsLeft.ts

```diff
@@ -0,0 +1,60 @@
+import { CurrentGameContext } from "contexts/CurrentGame"
+import { dateFromTimestamptzNow } from "lib/time"
+import { ActiveTurnPlayState } from "lib/turn"
+import useInterval from "lib/useInterval"
+import { last } from "lodash"
+import React from "react"
+import useServerDateOffset from "./useServerDateOffset"
+
+/**
+ * Milliseconds to delay in between timer updates.
+ *
+ * Intentional < 1s to mitigate setInterval drift.
+ *
+ * @see https://stackoverflow.com/q/985670/14386557
+ */
+const INTERVAL_DELAY = 100
+
+export default function useSecondsLeft(
+  activeTurnPlayState: ActiveTurnPlayState
+): number {
+  const serverDateOffset = useServerDateOffset()
```

Comment:
Open to suggestions on when – and how often we do this synchronization. It currently happens once when the play state renders.